File: pastey-ios/pastey/pastey/Models/Entry.swift
//
//  Entry.swift
//  pastey
//
//  Created by Burak Duruk on 2024-06-11.
//

import Foundation

// MARK: - Entry
struct Entry: Codable, Identifiable {
    let id: Int
    let entryID: String
    let userID, fromDeviceID, toDeviceID: Int
    let encryptedData, createdAt, fromDeviceName: String

    enum CodingKeys: String, CodingKey {
        case id
        case entryID = "entry_id"
        case userID = "user_id"
        case fromDeviceID = "from_device_id"
        case toDeviceID = "to_device_id"
        case encryptedData = "encrypted_data"
        case createdAt = "created_at"
        case fromDeviceName = "from_device_name"
    }
}

typealias EntryResponse = [Entry]


// MARK: - CopyRequest
struct CopyRequest: Codable {
    let fromDeviceID: Int
    let copies: [Copy]

    enum CodingKeys: String, CodingKey {
        case fromDeviceID = "from_device_id"
        case copies
    }
}

// MARK: - Copy
struct Copy: Codable {
    let toDeviceID: Int
    let encryptedData: String

    enum CodingKeys: String, CodingKey {
        case toDeviceID = "to_device_id"
        case encryptedData = "encrypted_data"
    }
}

File: pastey-wails/backend/models/entry.go
package models

import "time"

type Entry struct {
	ID             int64     `json:"id"`
	EntryID        string    `json:"entry_id"`
	UserID         int64     `json:"user_id"`
	FromDeviceID   int64     `json:"from_device_id"`
	ToDeviceID     int64     `json:"to_device_id"`
	EncryptedData  string    `json:"encrypted_data"`
	CreatedAt      time.Time `json:"created_at"`
	FromDeviceName string    `json:"from_device_name"`
}

type CopyRequest struct {
	FromDeviceID int64  `json:"from_device_id"`
	Copies       []Copy `json:"copies"`
}

type Copy struct {
	ToDeviceID    int64  `json:"to_device_id"`
	EncryptedData string `json:"encrypted_data"`
}
